contentcuration/contentcuration/frontend/administration/pages/Users/__tests__/emailUsersDialog.spec.js

```diff
@@ -46,128 +95,190 @@ function makeBulkWrapper() {
-      expect(wrapper.emitted('input')[0][0]).toBe(false);
-    });
+      await user.click(screen.getByText('Send email'));
 
-    it('should show warning if subject is not blank', async () => {
-      await wrapper.setData({ subject: 'subject' });
-      await wrapper.findComponent('[data-test="cancel"]').trigger('click');
-      expect(wrapper.vm.showWarning).toBe(true);
+      expect(screen.getAllByText('Field is required').length).toBeGreaterThan(0);
+      expect(mockActions.sendEmail).not.toHaveBeenCalled();
     });
 
-    it('should show warning if message is not blank', async () => {
-      await wrapper.setData({ message: 'message' });
-      await wrapper.findComponent('[data-test="cancel"]').trigger('click');
-      expect(wrapper.vm.showWarning).toBe(true);
-    });
+    it('does not submit with missing subject or message', async () => {
+      const user = userEvent.setup();
+      renderComponent({ initialRecipients: [userId] });
 
-    it('confirming close should reset fields and close dialog', async () => {
-      await wrapper.setData({ subject: 'subject', message: 'message' });
-      wrapper.findComponent('[data-test="confirm"]').vm.$emit('confirm');
-      expect(wrapper.vm.subject).toBe('');
-      expect(wrapper.vm.message).toBe('');
-      expect(wrapper.emitted('input')[0][0]).toBe(false);
-    });
-  });
+      const messageInput = screen.getByLabelText(/email body/i);
+      const subjectInput = screen.getByLabelText(/subject line/i);
 
-  describe('on submit', () => {
-    let sendEmail;
+      await user.type(messageInput, 'Test Message');
+      await user.click(screen.getByText('Send email'));
+      expect(mockActions.sendEmail).not.toHaveBeenCalled();
 
-    beforeEach(() => {
-      sendEmail = jest.spyOn(wrapper.vm, 'sendEmail').mockReturnValue(Promise.resolve());
+      await user.clear(messageInput);
+      await user.type(subjectInput, 'Test Subject');
+      await user.click(screen.getByText('Send email'));
+      expect(mockActions.sendEmail).not.toHaveBeenCalled();
     });
+  });
 
-    it('should not send if subject is empty', async () => {
-      await wrapper.setData({ message: 'message' });
-      await wrapper.findComponent('[data-test="send"]').trigger('click');
-      expect(sendEmail).not.toHaveBeenCalled();
-    });
+  describe('placeholders', () => {
+    it('adds placeholder text to message when clicked', async () => {
+      const user = userEvent.setup();
+      renderComponent({ initialRecipients: [userId] });
 
-    it('should not send if message is empty', async () => {
-      await wrapper.setData({ subject: 'subject' });
-      await wrapper.findComponent('[data-test="send"]').trigger('click');
-      expect(sendEmail).not.toHaveBeenCalled();
+      const messageInput = screen.getByLabelText(/email body/i);
+      await user.type(messageInput, 'Hello ');
+      await user.click(screen.getByText('First name'));
+
+      expect(messageInput.value).toContain('Hello');
```

Comment:
In this test case you attempt to test for `'adds placeholder text to message when clicked'`, and you correctly clicked `First name` button, but this `expect` doesn't test that the placeholder is there.